source/blog/2018-07-19-ovn-dpdk.html.md

```diff
@@ -0,0 +1,89 @@
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. Empty strings represent kernel's default. non-DPDK compatible drivers may be set; NICs not using DPDK compatible drivers won't be configured for OVS-DPDK usage and won't have their CPU isolated in multiple NUMA environments.
+
+Additionally, there are several optional performance related variables:
+* `pmd_threads_count` -- determines the number of PMD threads per NIC (default: `1`).
+* `nr_2mb_hugepages` -- determines the number of 2MB hugepages, if 2MB hugepages are to be used (default: `1024`).
+* `nr_1gb_hugepages` -- determines the number of 1GB hugepages, if 1GB hugepages are to be used (default: `4`).
+* `use_1gb_hugepages` -- determines whether 1GB hugepages should be used, where 1GB hugepages are supported (default: `true`).
+
+For additional information, refer to the role's [repository](https://github.com/ovirt/ovirt-ansible-dpdk-setup/).
+
+### OVN Integration
+DPDK in oVirt now leverages the capabilities of OVN in the form of an OVN localnet. This allows for seamless connectivity across OVN networks, benefiting from OVN's software defined routers, switches, security groups, and ACL's.
+
+For more information about OVN localnet integration in oVirt, refer to oVirt's [Provider Physical Network RFE](https://ovirt.org/develop/release-management/features/network/provider-physical-network/).
+
+# Usage in oVirt
+1) Install `oVirt.dpdk-setup` Ansible role via ansible-galaxy:
+    ```
+    # ansible-galaxy install oVirt.dpdk-setup
+    ```
+2) Execute the role as described above.
+3) Unless the host was already configured, the host will restart for kernel changes to be applied. Note that after reboot the following values are changed (If they are not, you may have incompatible hardware, or you should report a bug on ovirt@users.org):
+    * IOMMU: ```/sys/devices/virtual/iommu/<DMAR device>/devices/<PCI address>``` should exist. DMAR devices are typically annotated by dmar0, dmar1, and so forth.
+    * hugepages: ```grep Huge /proc/meminfo``` should reflect the desired state as defined in the Ansible playbook (i.e. `Hugepagesize` and `HugePages_Total`)
+    * CPU partitioning: based on the devices that are to be used with a DPDK compatible driver, CPUs should be partitioned separately between each NUMA node. Refer to `lscpu` to see live CPU NUMA separation information, e.g.: `NUMA node1 CPU(s):     8-15`
+
+    Note: NICs using userspace drivers do not appear in the kernel; `ip` commands won't list devices using userspace drivers. in oVirt, such devices appear as `dpdk0`, `dpdk1`, and so forth.
+4) Create an oVirt network ("phys-net-demo") on your data center and attach it to your cluster.
+![](https://www.ovirt.org/images/create_phys_net.png)
+5) Attach "phys-net-demo" to DPDK NICs accross your cluster.
+![](https://www.ovirt.org/images/set_phys_net.png)
```

Comment:
it is horrible that we dump all images in a single directory. At least, make sure that all your dpdk images have a dpdk_ prefix, so it would be clearer to understand where they belong.

Reply:
@leongold shouldn't you add an "apology" about SELinux, and point to the relevant bug?